ui/app/components/clients/history.js

```diff
@@ -171,9 +154,13 @@ export default class History extends Component {
         this.startTimeFromResponse = response.formattedStartTime;
         this.endTimeFromResponse = response.formattedEndTime;
       }
-      // compare if the response and what you requested are the same. If they are not throw a warning.
-      // this only gets triggered if the data was returned, which does not happen if the user selects a startTime after for which we have data. That's an adapter error and is captured differently.
-      if (!this.areArraysTheSame(this.startTimeFromResponse, this.startTimeRequested)) {
+      // compare if the response startTime comes after the requested startTime. If true throw a warning.
+      if (
+        isAfter(
+          new Date(this.getActivityResponse.startTime),
+          new Date(this.startTimeRequested[0], this.startTimeRequested[1])
+        )
+      ) {
         this.responseRangeDiffMessage = `You requested data from ${month} ${year}. We only have data from ${this.startTimeDisplay}, and that is what is being shown here.`;
```

Comment:
In the screenshot I selected August as a new **end date** from the widget and it displayed as `7 2021` (the warning shouldn't even be displaying the end date, right?). Figure it's mirage weirdness - when I tested with the "real" mock data all seemed to update and query correctly! Just pointing out in case it hints to a larger issue I'm not aware of
<img width="766" alt="Screen Shot 2022-02-09 at 4 29 10 PM" src="https://user-images.githubusercontent.com/68122737/153313709-fb47da09-0b74-41e3-95db-69d4a116135b.png">

Reply:
Good catch. I needed another conditional to only trigger comparison when they've selected a startTime. Amended.